### PR TITLE
Rakesh user home page

### PR DESCRIPTION
**Pull request by SyntaxArchitect**
Added Navbar and Sidebar
Horizontal Scroll Hooks Function
ServiceCard Component & Cards
AdsComponent
BannerSlider & ServiceCategory Section
ServicesSection & TopServiceProviders
Homepage Setup

### CODE REVIEW

File: src/components/Layout/index.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
+// Layout.js
+import NavBar from '../../components/NavBar';
```
Comment (JaseelaAlimahamed) [inline src/components/Layout/index.jsx]: change the index.jsx name
if it is togglar function then you can give name like that , make the first letter should be capital of your file

File: src/components/NavBar.jsx [hunk 2]
```diff
@@ -0,0 +1,135 @@
+import React, { useState, useEffect } from 'react';
```
Comment (JaseelaAlimahamed) [inline src/components/NavBar.jsx]: make this component in lay out folder

File: src/components/SearchBar.jsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import React from "react";
```
Comment (JaseelaAlimahamed) [inline src/components/SearchBar.jsx]: make this in layout folder

File: src/components/SideBar.jsx [hunk 4]
```diff
@@ -2,6 +2,7 @@ import React from 'react';
 import { FaSignOutAlt } from 'react-icons/fa';
```
Comment (JaseelaAlimahamed) [inline src/components/SideBar.jsx]: y more side bar make it a file

File: src/components/hooks/useHorizontalScroll.js
Comment (JaseelaAlimahamed) [inline src/components/hooks/useHorizontalScroll.js:1]: hooks folder set in userhome component if it is use in  home page

File: src/components/reUsableComponents/ServiceCardComponent.jsx
Comment (JaseelaAlimahamed) [inline src/components/reUsableComponents/ServiceCardComponent.jsx:1]: make the path in user home resuable component

File: src/components/Layout/index.jsx
Comment (JaseelaAlimahamed) [inline src/components/Layout/index.jsx:1]: change name index

File: src/components/reUsableComponents/BottomNavBar.jsx
Comment (JaseelaAlimahamed) [inline src/components/reUsableComponents/BottomNavBar.jsx:1]: what you mean by bottom navbar?

File: src/components/reUsableComponents/homeReusuableComponents/ServiceCardComponent.jsx
Comment (JaseelaAlimahamed) [inline src/components/reUsableComponents/homeReusuableComponents/ServiceCardComponent.jsx:1]: is these all reusable component

File: src/components/user/homePageComponents/ServicesCategorySection.jsx [hunk 5]
```diff
@@ -1,69 +1,63 @@
...
-//     { id: 7, imgUrl: "https://5.imimg.com/data5/ANDROID/Default/2022/4/SJ/LI/OB/148640419/product-jpeg-500x500.jpg", category: 'Driver  ' },
-//     { id: 2, imgUrl: "https://www.softedemy.com/wp-content/uploads/2022/07/Electrical-Optimized.jpg", category: 'Electrical' },
-//     { id: 3, imgUrl: "https://s3-us-west-2.amazonaws.com/hiration/ghost/2021/12/images2895-5d26590a000bf-1024x683-compressed.jpg", category: 'Carpentry' },
-//     { id: 4, imgUrl: "https://content.jdmagicbox.com/v2/comp/mumbai/h3/022pxx22.xx22.231107164034.p9h3/catalogue/l-t-labour-and-manpower-supplier-dahisar-west-mumbai-labour-contractors-5falo0j56j.jpg", category: 'Construction' },
-//     { id: 5, imgUrl: "https://www.ziprecruiter.com/svc/fotomat/public-ziprecruiter/cms/510968128CleaningTechnician.jpg=ws1280x960", category: 'Cleaning' },
-// ]   
+// Sample data for service categories
+const serviceCategories = [
```
Comment (JaseelaAlimahamed) [inline src/components/user/homePageComponents/ServicesCategorySection.jsx]: mkake this array in diff component

File: src/components/user/homePageComponents/hooks/useHorizontalScroll.js
Comment (JaseelaAlimahamed) [inline src/components/user/homePageComponents/hooks/useHorizontalScroll.js:1]: did you give in different folder in multiple ?

File: src/pages/HomePage/index.jsx
Comment (JaseelaAlimahamed) [inline src/pages/HomePage/index.jsx:1]: avoid name index.jsx

File: src/pages/ProviderProfilePage/index.jsx
Comment (JaseelaAlimahamed) [inline src/pages/ProviderProfilePage/index.jsx:1]: avoid name index.jsx

File: src/pages/ProviderProfilePage/index.jsx [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+                                mediaCount: 21,
+                                servicesListed: 2,
+                                imageUrl: 'https://randomuser.me/api/portraits/men/75.jpg',
+                            },
+                            aboutDescription: `
+                Graphic Design is now a popular profession. It offers various opportunities in the fields of marketing, advertising, web design, and many more.
+                Whether you're working on digital media or print, understanding design principles is key. Beyond the aesthetic appeal, graphic designers must also
+                focus on delivering a message effectively. The profession is challenging but rewarding, as it allows individuals to use both creativity and strategy.
+              `,
+                            servicesData: [
```
Comment (JaseelaAlimahamed) [inline src/pages/ProviderProfilePage/index.jsx]: if data is big make it in a different component

File: src/routes/UserRoutes.jsx
Comment (JaseelaAlimahamed) [inline src/routes/UserRoutes.jsx:1]: layouts st in layout folder not in the user router

File: src/components/reUsableComponents/BottomNavBar.jsx
Comment (SyntaxArchitect) [reply src/components/reUsableComponents/BottomNavBar.jsx:1]: image

File: src/components/reUsableComponents/homeReusuableComponents/ServiceCardComponent.jsx
Comment (SyntaxArchitect) [reply src/components/reUsableComponents/homeReusuableComponents/ServiceCardComponent.jsx:1]: Currently, it's only used on the home page, but I have created it as a reusable card component.

File: src/components/Layout/index.jsx
Comment (SyntaxArchitect) [reply src/components/Layout/index.jsx:1]: Updated the file name from index.jsx to Layout.jsx

File: src/pages/HomePage/index.jsx
Comment (SyntaxArchitect) [reply src/pages/HomePage/index.jsx:1]: Updated the file name from index.jsx to Home.jsx

File: src/pages/ProviderProfilePage/index.jsx
Comment (SyntaxArchitect) [reply src/pages/ProviderProfilePage/index.jsx:1]: Updated the file name from index.jsx to ProviderProfile.jsx

File: src/components/user/homePageComponents/ServicesCategorySection.jsx (hunk 5 above)
Comment (SyntaxArchitect) [reply src/components/user/homePageComponents/ServicesCategorySection.jsx]: I’ve created the array in user -> homePageComponents -> mockData -> ServiceCategoriesData.js as requested and updated the necessary imports.

File: src/pages/ProviderProfilePage/index.jsx (hunk 6 above)
Comment (SyntaxArchitect) [reply src/pages/ProviderProfilePage/index.jsx]: I’ve updated the code as suggested. I added separate hooks with the fetch data logic and imported useProviderProfileData from components/user/ProviderProfilePageComponents/hooks/useProviderProfileData

File: src/routes/UserRoutes.jsx
Comment (SyntaxArchitect) [reply src/routes/UserRoutes.jsx:1]: Removed the nav-layout and its components from UserRoutes.jsx as Nikhil is handling this separately. Updated the structure accordingly.

**Review by JaseelaAlimahamed — APPROVED** [review]

**Pull request merged**